### PR TITLE
[#42] add search by programming language to Communities#index

### PR DESCRIPTION
**Pull request by jmkoni**
Add DatabaseCleaner to both profile and user spec to clear out DB between tests.
Moved mentee/mentor to Profile since Communities#index was primarily returning profiles
and because the search was on Profile, not User. Added pg_search gem to allow for text
search since, currently, languages are a text field, not a drop down selection.

### CODE REVIEW

File: app/views/profiles/_community_profile.erb [hunk 1]
```diff
@@ -6,25 +6,25 @@
 <div class="row">
   <div class="col-md-8 col-md-offset-2">
 
-    <% if users.empty? %>
-      <div class='alert alert-warning text-center'>No available mentor/mentee at the moment. Kindly check back later</div>
+    <% if profiles.empty? %>
+      <div class='alert alert-warning text-center'>No available <%= community_type.downcase %> at the moment. Kindly check back later</div>
```
Comment (SundayAdefila) [inline app/views/profiles/_community_profile.erb]: Cool

File: spec/models/profile_model_spec.rb [hunk 2]
```diff
@@ -89,4 +89,66 @@
       end
     end
   end
+
+  describe "ClassMethods" do
```
Comment (SundayAdefila) [inline spec/models/profile_model_spec.rb]: 👍

File: spec/models/profile_model_spec.rb [hunk 3]
```diff
@@ -89,4 +89,66 @@
...
+    before :all do
+      DatabaseCleaner.start
```
Comment (SundayAdefila) [inline spec/models/profile_model_spec.rb]: I think we can have this set up globally in the spec helper? that will ensure it's available for all specs.

File: Gemfile [hunk 4]
```diff
@@ -33,6 +34,7 @@ end
 
 group :test do
   gem 'shoulda-matchers', '~> 3.1'
+  gem 'database_cleaner'
```
Comment (SundayAdefila) [inline Gemfile]: 👍

File: spec/models/user_model_spec.rb [hunk 5]
```diff
@@ -47,6 +47,7 @@
 
   describe "ClassMethods" do
     before :all do
+      DatabaseCleaner.start
```
Comment (SundayAdefila) [inline spec/models/user_model_spec.rb]: Same as for profile spec, moving to spec helper will help to prevent repetition.

File: spec/models/profile_model_spec.rb (hunk 3 above)
Comment (jmkoni) [inline spec/models/profile_model_spec.rb]: it is available for all specs. But you do have to set it up in before :all/after :all individually.

File: spec/models/user_model_spec.rb (hunk 5 above)
Comment (jmkoni) [inline spec/models/user_model_spec.rb]: Tried a few different ways that they recommended from the docs and none of them work with the test suite as it is written. I think that fully implementing this in the way that you want would require an overhaul of the test suite that is out of scope for this issue.

Comment (SundayAdefila) [inline spec/models/user_model_spec.rb]: Hmmm....

What I have in mind is having a single `DatabaseCleaner.start` and a single `DatabaseCleaner.end` hooked directly into the spec config... E.g as follow:

```
...
  config.before(:each) do
    DatabaseCleaner.start
  end

  config.after(:each) do
    DatabaseCleaner.clean
  end
...
```

Then we won't need to call it individually everywhere we need it again.... if just works for all specs

Comment (jmkoni) [inline spec/models/user_model_spec.rb]: As I said... doesn't work.

Comment (jmkoni) [inline spec/models/user_model_spec.rb]: It probably would work with bigger changes to the tests, but that's way out of scope.
